File: src/edu/JavaFxPidev/entities/Voiture.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package edu.JavaFxPidev.entities;

import java.util.Date;
import java.util.List;

/**
 *
 * @author Mouad
 */
public class Voiture {
    private int id;
    private String marque;
    private String typevoiture;
    private String matricule;
    private String modele;
    private Date datefabrication;
    private String typecarburant;
    private int kilometrage;
    private String etat;
    private List<Contrat> contrats;
    
    
     public void addContrat(Contrat contrat) {
        contrats.add(contrat);
    }

    public List<Contrat> getContrats() {
        return contrats;
    }

    public void setContrats(List<Contrat> contrats) {
        this.contrats = contrats;
    }
    public Voiture() {
    }

    public Voiture(String marque, String typevoiture, String matricule, String modele, Date datefabrication, String typecarburant, int kilometrage, String etat) {
        this.marque = marque;
        this.typevoiture = typevoiture;
        this.matricule = matricule;
        this.modele = modele;
        this.datefabrication = datefabrication;
        this.typecarburant = typecarburant;
        this.kilometrage = kilometrage;
        this.etat = etat;
    }

    public Voiture(int id, String marque, String typevoiture, String matricule, String modele, Date datefabrication, String typecarburant, int kilometrage, String etat) {
        this.id = id;
        this.marque = marque;
        this.typevoiture = typevoiture;
        this.matricule = matricule;
        this.modele = modele;
        this.datefabrication = datefabrication;
        this.typecarburant = typecarburant;
        this.kilometrage = kilometrage;
        this.etat = etat;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getMarque() {
        return marque;
    }

    public void setMarque(String marque) {
        this.marque = marque;
    }

    public String getTypevoiture() {
        return typevoiture;
    }

    public void setTypevoiture(String typevoiture) {
        this.typevoiture = typevoiture;
    }

    public String getMatricule() {
        return matricule;
    }

    public void setMatricule(String matricule) {
        this.matricule = matricule;
    }

    public String getModele() {
        return modele;
    }

    public void setModele(String modele) {
        this.modele = modele;
    }

    public Date getDatefabrication() {
        return datefabrication;
    }

    public void setDatefabrication(Date datefabrication) {
        this.datefabrication = datefabrication;
    }

    public String getTypecarburant() {
        return typecarburant;
    }

    public void setTypecarburant(String typecarburant) {
        this.typecarburant = typecarburant;
    }

    public int getKilometrage() {
        return kilometrage;
    }

    public void setKilometrage(int kilometrage) {
        this.kilometrage = kilometrage;
    }

    public String getEtat() {
        return etat;
    }

    public void setEtat(String etat) {
        this.etat = etat;
    }

    @Override
    public String toString() {
        return "Voiture{" + "id=" + id + ", marque=" + marque + ", typevoiture=" + typevoiture + ", matricule=" + matricule + ", modele=" + modele + ", datefabrication=" + datefabrication + ", typecarburant=" + typecarburant + ", kilometrage=" + kilometrage + ", etat=" + etat + '}';
    }
    
    
}

File: dev-clan-emna - Copie/src/Entity/Voiture.php
<?php

namespace App\Entity;

use App\Repository\VoitureRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\DBAL\Types\Types;
use Symfony\Component\Validator\Constraints\NotBlank;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Serializer\Annotation\Groups;


#[ORM\Entity(repositoryClass: VoitureRepository::class)]
class Voiture
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    #[Groups("voitures")]
    #[ApiProperty]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    #[Groups("voitures")]

    #[ApiProperty]
    #[Assert\NotBlank(message:"Le champ marque ne doit pas être vide.")]

    private ?string $marque = null;

    #[ORM\Column(length: 255)]
    #[Groups("voitures")]
    #[ApiProperty]
    private ?string $typevoiture = null;

    #[ORM\Column(length: 255)]
    #[Groups("voitures")]
    #[ApiProperty]

    private ?string $matricule = null;
    #[Assert\NotBlank(message:"Le champ matricule ne doit pas être vide.")]

    #[ORM\Column(length: 255)]
    #[Groups("voitures")]
    #[ApiProperty]

    private ?string $modele = null;
    #[Assert\NotBlank(message:"Le champ modele ne doit pas être vide.")]

    #[ORM\Column(type: Types::DATE_MUTABLE)]
    #[Groups("voitures")]
    #[Assert\NotBlank(message:"Le champ date de fabrication ne doit pas être vide.")]

    private ?\DateTimeInterface $datefabrication = null;

    #[ORM\Column(length: 255)]
    #[Groups("voitures")]

    private ?string $typecarburant = null;

    #[ORM\Column]
    #[Groups("voitures")]
    #[Assert\NotBlank(message:"Le champ kilometrage ne doit pas être vide.")]

    private ?int $kilometrage = null;

    #[ORM\Column(length: 255)]
    #[Groups("voitures")]
    #[Assert\NotBlank(message:"Le champ Etat ne doit pas être vide.")]

    private ?string $etat = null;

    #[ORM\OneToMany(mappedBy: 'voiture', targetEntity: Contrat::class, cascade: ["remove"])]
    
    private Collection $contrats;

    public function __construct()
    {
        $this->contrats = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getMarque(): ?string
    {
        return $this->marque;
    }

    public function setMarque(string $marque): self
    {
        $this->marque = $marque;

        return $this;
    }

    public function getTypevoiture(): ?string
    {
        return $this->typevoiture;
    }

    public function setTypevoiture(string $typevoiture): self
    {
        $this->typevoiture = $typevoiture;

        return $this;
    }

    public function getMatricule(): ?string
    {
        return $this->matricule;
    }

    public function setMatricule(string $matricule): self
    {
        $this->matricule = $matricule;

        return $this;
    }

    public function getModele(): ?string
    {
        return $this->modele;
    }

    public function setModele(string $modele): self
    {
        $this->modele = $modele;

        return $this;
    }

    public function getDatefabrication(): ?\DateTimeInterface
    {
        return $this->datefabrication;
    }

    public function setDatefabrication(\DateTimeInterface $datefabrication): self
    {
        $this->datefabrication = $datefabrication;

        return $this;
    }

    public function getTypecarburant(): ?string
    {
        return $this->typecarburant;
    }

    public function setTypecarburant(string $typecarburant): self
    {
        $this->typecarburant = $typecarburant;

        return $this;
    }

    public function getKilometrage(): ?int
    {
        return $this->kilometrage;
    }

    public function setKilometrage(int $kilometrage): self
    {
        $this->kilometrage = $kilometrage;

        return $this;
    }

    public function getEtat(): ?string
    {
        return $this->etat;
    }

    public function setEtat(string $etat): self
    {
        $this->etat = $etat;

        return $this;
    }

    /**
     * @return Collection<int, Contrat>
     */
    public function getContrats(): Collection
    {
        return $this->contrats;
    }

    public function addContrat(Contrat $contrat): self
    {
        if (!$this->contrats->contains($contrat)) {
            $this->contrats->add($contrat);
            $contrat->setVoiture($this);
        }

        return $this;
    }

    public function removeContrat(Contrat $contrat): self
    {
        if ($this->contrats->removeElement($contrat)) {
            // set the owning side to null (unless already changed)
            if ($contrat->getVoiture() === $this) {
                $contrat->setVoiture(null);
            }
        }

        return $this;
    }
}
